components/ai_chat/core/browser/engine/engine_consumer_conversation_api_v2_unittest.cc

```diff
@@ -1330,36 +1332,93 @@ TEST_F(EngineConsumerConversationAPIUnitTest, GenerateEvents_UploadImage) {
+      .WillOnce([&](std::vector<OAIMessage> messages,
                     const std::string& selected_language,
                     std::optional<base::Value::List> oai_tool_definitions,
                     const std::optional<std::string>& preferred_tool_name,
                     mojom::ConversationCapability conversation_capability,
                     EngineConsumer::GenerationDataCallback data_callback,
                     EngineConsumer::GenerationCompletedCallback callback,
                     const std::optional<std::string>& model_name) {
-        ASSERT_EQ(conversation.size(), 3u);
-        EXPECT_EQ(conversation[0].role, ConversationEventRole::kUser);
-        for (size_t i = 0; i < 3; ++i) {
+        // Should have 1 message
+        ASSERT_EQ(messages.size(), 1u);
+        EXPECT_EQ(messages[0].role, "user");
+
+        // Content blocks: images text + 3 images + screenshots text +
+        // 3 screenshots + prompt = 9
+        ASSERT_EQ(messages[0].content.size(), 9u);
+
+        // Check uploaded images text
+        ASSERT_EQ(messages[0].content[0]->which(),
+                  mojom::ContentBlock::Tag::kTextContentBlock);
+        EXPECT_EQ(messages[0].content[0]->get_text_content_block()->text,
+                  "These images are uploaded by the user");
+
+        // Check 3 uploaded images
+        for (size_t i = 1; i <= 3; ++i) {
+          ASSERT_EQ(messages[0].content[i]->which(),
+                    mojom::ContentBlock::Tag::kImageContentBlock);
           EXPECT_EQ(
-              GetContentStrings(conversation[0].content)[i],
-              base::StrCat({"data:image/png;base64,",
-                            base::Base64Encode(uploaded_images[i]->data)}));
+              messages[0].content[i]->get_image_content_block()->image_url,
+              GURL(EngineConsumer::GetImageDataURL(
+                  uploaded_images[i - 1]->data)));
         }
-        EXPECT_EQ(conversation[0].type, ConversationEventType::kUploadImage);
-        for (size_t i = 3; i < uploaded_images.size(); ++i) {
+
+        // Check screenshots text
+        ASSERT_EQ(messages[0].content[4]->which(),
+                  mojom::ContentBlock::Tag::kTextContentBlock);
+        EXPECT_EQ(messages[0].content[4]->get_text_content_block()->text,
+                  "These images are screenshots");
+
+        // Check 3 screenshots
+        for (size_t i = 5; i <= 7; ++i) {
+          ASSERT_EQ(messages[0].content[i]->which(),
+                    mojom::ContentBlock::Tag::kImageContentBlock);
           EXPECT_EQ(
-              GetContentStrings(conversation[1].content)[i - 3],
-              base::StrCat({"data:image/png;base64,",
-                            base::Base64Encode(uploaded_images[i]->data)}));
+              messages[0].content[i]->get_image_content_block()->image_url,
+              GURL(EngineConsumer::GetImageDataURL(
+                  uploaded_images[i - 2]->data)));
```

Comment:
can we do a separate image_idx? so we just do `uploaded_images[image_idx++]->data)` to avoid confusion.

Reply:
https://github.com/brave/brave-core/pull/32829/commits/d82a3e047517d819ecaec6f6c73d605e774d92ab